clouddriver-gae/src/main/groovy/com/netflix/spinnaker/clouddriver/gae/security/GaeCredentialsInitializer.groovy

```diff
@@ -0,0 +1,93 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.gae.security
+
+import com.netflix.spinnaker.cats.module.CatsModule
+import com.netflix.spinnaker.clouddriver.gae.config.GaeConfigurationProperties
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository
+import com.netflix.spinnaker.clouddriver.security.CredentialsInitializerSynchronizable
+import com.netflix.spinnaker.clouddriver.security.ProviderUtils
+import org.apache.log4j.Logger
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.stereotype.Component
+
+@Component
+@Configuration
+class GaeCredentialsInitializer implements CredentialsInitializerSynchronizable {
+  private static final Logger log = Logger.getLogger(this.class.simpleName)
+
+  @Autowired
+  String gaeApplicationName
+
+  @Autowired
+  AccountCredentialsRepository accountCredentialsRepository
+
+  @Bean
+  List<? extends GaeNamedAccountCredentials> gaeNamedAccountCredentials(
+    GaeConfigurationProperties gaeConfigurationProperties) {
+    synchronizeGaeAccounts(gaeConfigurationProperties, null)
+  }
+
+  @Override
+  String getCredentialsSynchronizationBeanName() {
+    return "synchronizeGaeAccounts"
+  }
+
+  List<?> synchronizeGaeAccounts(GaeConfigurationProperties gaeConfigurationProperties, CatsModule catsModule) {
+    def (ArrayList<GaeConfigurationProperties.ManagedAccount> accountsToAdd, List<String> namesOfDeletedAccounts) =
+      ProviderUtils.calculateAccountDeltas(accountCredentialsRepository,
+                                           GaeNamedAccountCredentials,
+                                           gaeConfigurationProperties.accounts)
+
+    accountsToAdd.each { GaeConfigurationProperties.ManagedAccount managedAccount ->
+      try {
+        def jsonKey = GaeCredentialsInitializer.getJsonKey(managedAccount)
+        def gaeAccount = new GaeNamedAccountCredentials.Builder()
+          .name(managedAccount.name)
+          .environment(managedAccount.environment ?: managedAccount.name)
+          .accountType(managedAccount.accountType ?: managedAccount.name)
+          .project(managedAccount.project)
+          .jsonKey(jsonKey)
+          .applicationName(gaeApplicationName)
+          .requiredGroupMembership(managedAccount.requiredGroupMembership)
+          .build()
+
+        if (!managedAccount.project) {
```

Comment:
This check should probably happen before calling the builder. Also why is this the only thing you check? Seems like the other fields are pretty important too.